src/types/acls.rs

```diff
@@ -0,0 +1,387 @@
+use crate::error::Error;
+use crate::parse::{parse_until_done, MapOrNot};
+use crate::types::UnsolicitedResponse;
+use imap_proto::types::AclRight;
+use imap_proto::Response;
+use ouroboros::self_referencing;
+use std::borrow::Cow;
+use std::collections::HashSet;
+use std::fmt::{Display, Formatter};
+use std::sync::mpsc;
+
+/// Specifies how [`Session::set_acl`] should modify an existing permission set.
```

Comment:
Clippy seems to think that this link doesn't work. Almost certainly because `Session` isn't in scope here. You can fix that by adding this to the `use` list:

```rust
#[cfg(doc)]
use crate::Session;
```

I think.